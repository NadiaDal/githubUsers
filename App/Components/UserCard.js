// flow
import React from 'react'
import {View, Image, Text} from 'react-native'
import styles from './Styles/UserCardStyles'

type UserProps = {
  item: Object
}

export default function UserCard (props: UserProps) {
  const {item} = props
  return (
    <View style={styles.profile}>
      <Image
        style={styles.avatar}
        source={{uri: item.avatar_url}}
      />
      <View>
        <Text style={styles.login}>{item.login}</Text>
        <Text
          numberOfLines={2}
          ellipsizeMode={'head'}
          style={styles.url}
        >{item.html_url}
        </Text>
      </View>
    </View>
  )
}
