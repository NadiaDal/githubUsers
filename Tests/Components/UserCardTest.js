import 'react-native'
import React from 'react'
import UserCard from '../../App/Components/UserCard'
import renderer from 'react-test-renderer'

test('UserCard component renders correctly', () => {
  const item = {
    avatar_url: 'https://avatars0.githubusercontent.com/u/2?v=4',
    html_url: 'https://github.com/defunkt',
    login: 'defunkt'
  }
  const tree = renderer.create(<UserCard item={item} />).toJSON()
  expect(tree).toMatchSnapshot()
})
