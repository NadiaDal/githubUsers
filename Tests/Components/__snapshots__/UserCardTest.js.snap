// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F7",
      "borderRadius": 10,
      "flexDirection": "row",
      "marginBottom": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 5,
      "padding": 10,
      "width": 730,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://avatars0.githubusercontent.com/u/2?v=4",
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 100,
        "marginRight": 10,
        "width": 100,
      }
    }
  />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgb(32, 125, 64)",
          "fontSize": 20,
        }
      }
    >
      defunkt
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="head"
      numberOfLines={2}
      style={Object {}}
    >
      https://github.com/defunkt
    </Text>
  </View>
</View>
`;
